tests/testTriggers.go

```diff
@@ -9005,6 +9175,132 @@ func TriggerOCPStorageNodeRecycle(contexts *[]*scheduler.Context, recordChan *ch
 
 }
 
+// TriggerReallocSharedMount peforms sharedv4 and sharedv4_svc volumes reallocation
+func TriggerReallocSharedMount(contexts *[]*scheduler.Context, recordChan *chan *EventRecord) {
+	defer ginkgo.GinkgoRecover()
+	defer endLongevityTest()
+	startLongevityTest(ReallocateSharedMount)
+	event := &EventRecord{
+		Event: Event{
+			ID:   GenerateUUID(),
+			Type: ReallocateSharedMount,
+		},
+		Start:   time.Now().Format(time.RFC1123),
+		Outcome: []error{},
+	}
+
+	defer func() {
+		event.End = time.Now().Format(time.RFC1123)
+		*recordChan <- event
+	}()
+	setMetrics(*event)
+	stepLog := "get nodes with shared mount and reboot them"
+	Step(stepLog, func() {
+		log.InfoD(stepLog)
+		for _, ctx := range *contexts {
+			vols, err := Inst().S.GetVolumes(ctx)
+			if err != nil {
+				UpdateOutcome(event, err)
+				continue
+			}
+
+			for _, vol := range vols {
+				if vol.Shared {
+
+					n, err := Inst().V.GetNodeForVolume(vol, 1*time.Minute, 5*time.Second)
+					if err != nil {
+						UpdateOutcome(event, err)
+						continue
+					}
+
+					log.InfoD("volume %s is attached on node %s [%s]", vol.ID, n.SchedulerNodeName, n.Addresses[0])
+
+					// Workaround to avoid PWX-24277 for now.
+					Step(fmt.Sprintf("wait until volume %v status is Up", vol.ID), func() {
+						log.InfoD("wait until volume %v status is Up", vol.ID)
+
+						t := func() (interface{}, bool, error) {
+							connOpts := node.ConnectionOpts{
+								Timeout:         1 * time.Minute,
+								TimeBeforeRetry: 5 * time.Second,
+								Sudo:            true,
+							}
+							cmd := fmt.Sprintf("pxctl volume inspect %s | grep \"Replication Status\"", vol.ID)
```

Comment:
How about we use InspectVolume?

Reply:
volumeinspect sometimes takes time to reflect the actual status, hence used cli here